vizro-core/src/vizro/models/_navigation/navigation.py

```diff
@@ -19,20 +21,23 @@ class Navigation(VizroBaseModel):
     Args:
         pages (Optional[NavigationPagesType]): See [`NavigationPagesType`][vizro.models.types.NavigationPagesType].
             Defaults to `None`.
+        selector (Optional[NavigationSelectorType])
     """
 
     pages: Optional[NavigationPagesType] = None
-    _selector: Accordion = PrivateAttr()
+    selector: Optional[NavigationSelectorType] = None
 
     # validators
     _validate_pages = validator("pages", allow_reuse=True)(_validate_pages)
 
     @_log_call
     def pre_build(self):
-        from vizro.models._navigation._accordion import Accordion
-
-        self._selector = Accordion(pages=self.pages)  # type: ignore[arg-type]
+        if self.selector is None:
+            self.selector = Accordion(pages=self.pages)
 
     @_log_call
     def build(self, *, active_page_id=None):
-        return self._selector.build(active_page_id=active_page_id)
+        if isinstance(self.selector, NavBar):
+            return self.selector.build(active_page_id=active_page_id)
+        if isinstance(self.selector, Accordion):
+            return None, self.selector.build(active_page_id=active_page_id)
```

Comment:
This has been partially addressed. Meaning, 'None' is removed, and instead in case of Accordion selector this is returned: 
        ```
if isinstance(self.selector, Accordion):
            return html.Div(
                children=[
                    html.Div(className="hidden", id="nav_bar_outer"),
                    self.selector.build(active_page_id=active_page_id),
                ]
            )
```
I still had to check for the selector type, because Accordion model returns only nav_panel_outer.

Reply:
Can't think of a better way right now either, so let's keep it like you have it right now 👍